moveit_core/trajectory_processing/src/time_optimal_trajectory_generation.cpp

```diff
@@ -891,12 +891,12 @@ bool TimeOptimalTrajectoryGeneration::computeTimeStamps(robot_trajectory::RobotT
     RCLCPP_ERROR(LOGGER, "Failed to get active variable indices.");
   }
 
-  const size_t num_joints = indices.size();
-  Eigen::VectorXd max_velocity(num_joints);
-  Eigen::VectorXd max_acceleration(num_joints);
-  for (const auto idx : indices)
+  const size_t num_active_joints = indices.size();
```

Comment:
So if I understand this correctly, you want to use `indices` when getting bounds, but the raw `idx` everywhere else?

Reply:
Yeah, I had originally messed this up, so here is the idea.

Suppose you have a robot with 5 real joints and 1 mimic joint: `indices` might give you something like `[0, 1, 3, 4]`, let's say, so index `2` is the mimic.

The limits then have to look through 4 joints (this is `num_active_joints`), and each element should correspond to the limits for joints `0`, `1`, `3`, and `4`, respectively.

tl;dr
* `indices = [0, 1, 3, 4]`
* `idx` in the loop takes values `[0, 1, 2, 3]`

Reply:
Thanks, that makes sense!